oolab/src/main/java/agh/ics/oop/model/Animal.java

```diff
@@ -0,0 +1,87 @@
+package agh.ics.oop.model;
+
+import java.util.Objects;
+
+public class Animal {
+    private MapDirection direction;
+    private Vector2d coordinate;
+
+    public Animal() {
+        this.direction = MapDirection.NORTH;
+        this.coordinate = new Vector2d(2, 2);
+    }
+    public Animal(Vector2d coordinate) {
+        this.direction = MapDirection.NORTH;
+        if (coordinate.getX() >4 || coordinate.getX() <-4 || coordinate.getY() >4 || coordinate.getY() <-4) {
```

Comment:
nie podoba mi się to, lepiej zrobić wektory określające narożniki i później sprawdzać czy pozycja jest w ich przedziale

Reply:
mamy do tego funkcje w vector2d